File: src/vpc/internal/entitlementsApply.js
"use strict";

const { verifyInternalSignature } = require("../../common/auth");

async function handleInternalEntitlementsApply({ event, method, res }) {
  if (method !== "POST") return res.methodNotAllowed("Method not allowed");

  const v = verifyInternalSignature(event);
  if (!v.ok) return res.forbidden(`Internal auth failed: ${v.reason}`);

  return res.notImplemented("TODO: internal entitlements apply (write RDS)");
}

module.exports = { handleInternalEntitlementsApply };

File: src_C/Vpc/Internal/EntitlementsApply.cs
using Amazon.Lambda.APIGatewayEvents;
using RecallSmith.Lambda.Common;

namespace RecallSmith.Lambda.Vpc.Internal;

public static class EntitlementsApply
{
  public static Task<APIGatewayProxyResponse> HandleInternalEntitlementsApply(LambdaRequest req, Res res)
  {
    if (req.Method != "POST") return Task.FromResult(res.MethodNotAllowed("Method not allowed"));

    var v = Auth.VerifyInternalSignature(req);
    if (!v.Ok) return Task.FromResult(res.Forbidden($"Internal auth failed: {v.Reason}"));

    return Task.FromResult(res.NotImplemented("TODO: internal entitlements apply (write RDS)"));
  }
}

